native/dgelqf.go

```diff
@@ -0,0 +1,81 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import (
+	"github.com/gonum/blas"
+	"github.com/gonum/lapack"
+)
+
+// Dgelqf computes the LQ factorization of the m×n matrix a using a blocked
+// algorithm. Please see the documentation for Dgelq2 for a description of the
+// parameters at entry and exit.
+//
+// work is vector of temporary storage with length at least m, and will panic if
+// work is shorter. Lwork is a parameter specifying the accessible length of the
+// work slice. Dgelqf is a blocked LQ factorization, but the block size is limited
+// by the temporary space available. If lwork == -1, instead of performing Dgelqf,
+// instead the optimal work length will stored into work[0].
```

Comment:
Yes, this is much better. But the wording is strange. It seems to me that 'instead' can be deleted and 'will stored' should be 'will be stored'.

From the previous reading I remember that this comment clarification could be also done for other functions below. Sorry for not poiting it out earlier.